src/lib.rs

```diff
@@ -130,3 +136,98 @@ pub fn generate_salt_suffix(deployer: &[u8], salt_suffix: &str, prefix: &str) ->
+            ("c", "70b556548FF0161082fB751d5E372eFa0133805C"),
+            // https://www.poetryfoundation.org/poems/44263/fire-and-ice
+            ("Some say the world will end in fire, Some say in ice. From what I’ve tasted of desire I hold with those who favor fire. But if it had to perish twice, I think I know enough of hate To say that for destruction ice Is also great And would suffice.", "C244c5dEa48e677cE7cAbD05BF8eC220b1a99Fc9")
+        ];
+
+        for (salt, answer) in correct_answers.iter() {
+            let addr: [u8; 20] = calc_addr(deployer, salt.as_bytes());
+            let addr_str = hex::encode(addr);
+            assert_eq!(addr_str, answer.to_lowercase());
+        }
+    }
+
+    #[test]
+    fn should_generate_with_prefix() {
+        let deployer: &Vec<u8> = &hex::decode("5e17b14ADd6c386305A32928F985b29bbA34Eff5").unwrap();
+        let runs = vec!["0", "00", "000", "abc", "123", "789"];
+
+        for run in runs.iter() {
+            let salt = generate_salt(deployer, run);
+            
+            // NOTE:    would rather the library expose a function. Waiting until major release to do so
+            //          let addr: [u8; 20] = calc_addr_with_bytes(deployer, &salt);
+            let mut bytes: Vec<u8> = Vec::new();
+            bytes.push(0xff);
+            bytes.extend_from_slice(deployer);
+            bytes.extend_from_slice(&salt);
+            bytes.extend_from_slice(&KECCAK256_PROXY_CHILD_BYTECODE);
+            let hash = Keccak256::digest(&bytes);
+            let mut proxy_bytes = [0u8; 20];
+            proxy_bytes.copy_from_slice(&hash[12..]);
+        
+            // Use proxy address to compute the final contract address.
+            // keccak256(rlp(proxy_bytes ++ 0x01)) More here -> https://ethereum.stackexchange.com/a/761/66849
+            let mut bytes2: Vec<u8> = Vec::new();
+            bytes2.extend_from_slice(&[0xd6, 0x94]); // RLP prefix for a list of two items
+            bytes2.extend(&proxy_bytes); // The proxy address
+            bytes2.push(0x01); // The nonce of the contract
+            let hash2 = Keccak256::digest(&bytes2);
+        
+            // resulting hash -> The last 20 bytes (40 characters) of the hash.
+            let mut address = [0u8; 20];
+            address.copy_from_slice(&hash2[12..]);
+
+            assert!(hex::encode(address).starts_with(run));
+        }
+    }
+
+    #[test]
+    fn should_generate_with_empty_prefix() {
+        let deployer: &Vec<u8> = &hex::decode("0fC5025C764cE34df352757e82f7B5c4Df39A836").unwrap();
+        generate_salt(deployer, "");
+    }
+
+    #[test]
+    fn should_generate_with_salt_prefix() {
+        let deployer: &Vec<u8> = &hex::decode("5e17b14ADd6c386305A32928F985b29bbA34Eff5").unwrap();
+        let runs = vec!["0", "00", "000", "abc", "123", "789"];
+        let salt_prefix = "testpfx_";
+        for run in runs.iter() {
+            generate_salt_suffix(deployer, salt_prefix, run);
+        }
```

Comment:
here we should check if the generated address is actually starting with the same values or not.

For this either have to modify the method `generate_salt_suffix` to return the value of new address or can use the `calc_addr` and check if the address is staring with that characters or not.

Reply:
Will need to modify the `generate_salt_suffix` to return the new address, which I tried to avoid for this PR. I'll introduce some more changes with the next commit that may lead to a major release